pkg/auth/providers/azure/azure_provider.go

```diff
@@ -598,3 +574,130 @@ func (ap *azureProvider) GetUserExtraAttributes(userPrincipal v3.Principal) map[
 	}
 	return extras
 }
+
+func (ap *azureProvider) loginUserAzureADGraph(credential *v32.AzureADLogin, config *v32.AzureADConfig, test bool) (v3.Principal, []v3.Principal, string, error) {
+	logrus.Debug("[AZURE_PROVIDER] Started token swap with AzureAD")
+
+	azureClient, err := adclient.NewClientFromCode(credential.Code, config)
+	if err != nil {
+		return v3.Principal{}, nil, "", err
+	}
+
+	logrus.Debug("[AZURE_PROVIDER] Completed token swap with AzureAD")
+
+	oid, err := extractFieldFromJWT(azureClient.AccessToken(), "oid")
+	if err != nil {
+		return v3.Principal{}, nil, "", err
+	}
+
+	logrus.Debug("[AZURE_PROVIDER] Started getting user info from AzureAD")
+
+	u, err := azureClient.GetUser(oid.(string))
```

Comment:
With this always be safe? To ensure safety you can do something like `id, _ := oid.(string)` above instead.